sdv/multi_table/base.py

```diff
@@ -161,3 +161,64 @@ def validate(self, data):
 
         if errors:
             raise InvalidDataError(errors)
+
+    def auto_assign_transformers(self, data):
+        """Automatically assign the required transformers for the given data and constraints.
+
+        This method will automatically set a configuration to the ``rdt.HyperTransformer``
+        with the required transformers for the current data.
+
+        Args:
+            data (dict):
+                Mapping of table name to pandas.DataFrame.
+
+        Raises:
+            InvalidDataError:
+                If a table of the data is not present in the metadata.
+        """
+        for table_name, table_data in data.items():
+            if table_name not in self._table_synthesizers:
+                raise InvalidDataError([f"Table '{table_name}' is not present in the metadata."])
+
+            self._table_synthesizers[table_name].auto_assign_transformers(table_data)
```

Comment:
We can use `self.validate(data)` on the first lane, then we remove the `if - raise` statement.
Which leads me to this question:
Should we move `validate` as a decorator, which will allow us to do this (Its fine to just call `self.validate` but it's fancier I guess). @amontanez24 @fealho 
```
@validate
def method_that_uses_data(self, data):
   # logic that doesn't need to handle anything from the data
```

Reply:
Seems reasonable to me.

Reply:
yeah I think it's fine, I just don't know if we need to do all the validation every time